drivers/i2c/i2c_esp32.c

```diff
@@ -272,6 +294,7 @@ static int i2c_esp32_configure(const struct device *dev, uint32_t dev_config)
 	}
 
 	i2c_hal_set_bus_timing(&data->hal, config->bitrate, i2c_get_clk_src(config->bitrate));
+	i2c_esp32_configure_timeout(dev);
```

Comment:
I will review this, but, have checked `i2c_hal_set_tout()` function? It sets the i2C timeout and might be used in here instead.

Reply:
Yes, I did, but it does not allow to completely disable the timeout function using the `I2C_TIME_OUT_EN` bit mentioned in the reference manual. My initial goal was to use that bit to disable it by default. Unfortunately, this did not have any effect during my tests.

Also the HAL function does not perform any checks whether the value is too large or not. And it expects the value in number of clocks format instead of milliseconds or another time unit. So I decided to access the registers directly, as we need to get the maximum size of the value from the reg define and the clock frequency by other means anyways.